### PR TITLE
Change instrument to include TimeROI in calculation

### PR DESCRIPTION
**Pull request by ghost**
**Description of work.**

Change instrument to include TimeROI in calculation of statistics.

**To test:**

All existing tests should pass.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v6.7.0/Direct_Geometry/General/New_features/35127.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+- `XMLInstrumentParameter` no includes TimeROI in the calculation of instrument parameters.
```
Comment (peterfpeterson) [inline docs/source/release/v6.7.0/Direct_Geometry/General/New_features/35127.rst:1]:
```suggestion
- `XMLInstrumentParameter` now includes TimeROI in the calculation of instrument parameters.
```

File: Framework/Geometry/test/XMLInstrumentParameterTest.h [hunk 2]
```diff
@@ -50,42 +51,48 @@ class XMLInstrumentParameterTest : public CxxTest::TestSuite {
...
+    roi->addROI("2000-11-30T01:01:01", "2000-11-30T01:01:02");
     TimeSeriesProperty<double> series("doubleProperty");
 
     const double expectedFilteredValue = 1;
     series.addValue("2000-11-30T01:01:01", 0);
     series.addValue("2000-11-30T01:01:02", expectedFilteredValue);
 
     XMLInstrumentParameter_sptr logFile = make_logfile_object("last_value");
-    const double actualFilteredValue = logFile->createParamValue(&series);
+    const double actualFilteredValue = logFile->createParamValue(&series, roi);
```
Comment (jmborr) [inline Framework/Geometry/test/XMLInstrumentParameterTest.h:89]: can you add additional test with different ROI's?

roi->addROI("2000-11-30T01:01:00", "2000-11-30T01:01:01");
In this case, actualFilteredValue should be 0

File: Framework/Geometry/test/XMLInstrumentParameterTest.h [hunk 3]
```diff
@@ -140,13 +153,14 @@ class XMLInstrumentParameterTest : public CxxTest::TestSuite {
...
     XMLInstrumentParameter_sptr logFile = make_logfile_object("median");
-    const double actualFilteredValue = logFile->createParamValue(&series);
+    const double actualFilteredValue = logFile->createParamValue(&series, roi);
     TSM_ASSERT_EQUALS("Filtering by Median is not performed correctly", expectedFilteredValue, actualFilteredValue);
   }
 
   // This functionality will soon be legacy, since filtering by nth-position is
   // not a good idea.
   void test_filter_by_nth_position() {
+    Kernel::TimeROI *roi = nullptr;
```
Comment (jmborr) [inline Framework/Geometry/test/XMLInstrumentParameterTest.h:163]: I think that if the ROI excludes the first value, `position 2` should correspond to the value `2`. Can you please test with ROI `[02, 04]`?

**Review by jmborr — CHANGES_REQUESTED** [review]

File: Framework/Geometry/test/XMLInstrumentParameterTest.h [hunk 4]
```diff
@@ -50,42 +51,48 @@ class XMLInstrumentParameterTest : public CxxTest::TestSuite {
...
     XMLInstrumentParameter_sptr logFile = make_logfile_object("first_value");
-    const double actualFilteredValue = logFile->createParamValue(&series);
+    const double actualFilteredValue = logFile->createParamValue(&series, roi);
     TSM_ASSERT_EQUALS("Filtering by First Value is not performed correctly", expectedFilteredValue,
                       actualFilteredValue);
   }
 
   void test_filter_by_last_value() {
+    Kernel::TimeROI *roi = new Kernel::TimeROI;
+    roi->addROI("2000-11-30T01:01:01", "2000-11-30T01:01:02");
```
[the remark this answers is not in the document]
Comment (searscr) [reply Framework/Geometry/test/XMLInstrumentParameterTest.h]: Filter by last value uses the nth value, TimeROI is not yet taken into account.

**Review by jmborr — APPROVED** [review]

**Pull request merged**